src/modules/apps/e2e/index.ts

```diff
@@ -0,0 +1,118 @@
+import chalk from 'chalk'
+
+import { createAppsClient } from '../../../lib/clients/IOClients/infra/Apps'
+import { Tester, TestRequest, AppReport } from '../../../lib/clients/IOClients/apps/Tester'
+import { ManifestEditor } from '../../../lib/manifest/ManifestEditor'
+import { SessionManager } from '../../../lib/session/SessionManager'
+import { parseReport, AppTest, passedApp } from './specsState'
+import { parseLocator } from '../../../locator'
+import logger from '../../../logger'
+import { clusterIdDomainInfix, publicEndpoint } from '../../../env'
+
+const POLL_INTERVAL = 2000
+
+class EndToEndCommand {
+  private tester: Tester
+  constructor(private options) {
+    this.tester = Tester.createClient()
+  }
+
+  public run() {
+    if (this.options.workspace) {
+      return this.runWorkspaceTests()
+    }
+
+    return this.runAppTests()
+  }
+
+  private async runAppTests() {
+    const manifestEditor = await ManifestEditor.getManifestEditor()
+    const cleanAppId = manifestEditor.appLocator
+
+    const apps = createAppsClient()
+    const { data: workspaceAppsList } = await apps.listApps()
+    const appItem = workspaceAppsList.find(({ app }) => app.startsWith(cleanAppId))
+
+    if (appItem === undefined) {
+      throw new Error(`App "${cleanAppId}" was not found in the current workspace!`)
+    }
+
+    const testRequest = this.options.report
+      ? null
+      : await this.tester.test(
+          {
+            integration: true,
+            monitoring: true,
+            authToken: this.options.token ? SessionManager.getSingleton().token : undefined,
+          },
+          appItem.id
+        )
+
+    this.watch(testRequest)
+  }
+
+  private workspaceBaseUrl() {
+    const { account, workspace } = SessionManager.getSingleton()
+    return `https://${workspace}--${account}${clusterIdDomainInfix()}.${publicEndpoint()}`
+  }
+
+  private reportUrl(testId: string) {
+    return `${this.workspaceBaseUrl()}/_v/report/${testId}/html`
+  }
+
+  private async runWorkspaceTests() {
+    const testRequest = this.options.report
+      ? null
+      : await this.tester.test({
+          integration: true,
+          monitoring: true,
+          authToken: this.options.token ? SessionManager.getSingleton().token : undefined,
+        })
+
+    this.watch(testRequest)
```

Comment:
Just to make sure someone will handle this promise result:
```suggestion
    return this.watch(testRequest)
```